### PR TITLE
Searching mailling lists by name

### PR DESCRIPTION
**Pull request by italopaiva**
The user now can search his/her mailling lists by name. 

Same as [Pull Request 187](https://github.com/colab/colab/pull/187).

### CODE REVIEW

File: src/colab_superarchives/views.py [hunk 1]
```diff
@@ -418,43 +419,86 @@ def get_object(self, *args, **kwargs):
...
-    def get_context_data(self, **kwargs):
+            response = redirect('user_profile', username=user.username)
+
+        return response
+
+    def search_lists_by_name(self, user, listname):
+        context = self.get_mailling_lists(user, listname)
+
+        html = render_to_string(u'accounts/subscription_lists.html',
+                                {'membership': context['membership']})
```
Comment (chaws) [inline src/colab_superarchives/views.py:456]: @italopaiva , I had to add 'per_page' so the pagination would work with search by list name: https://github.com/colab/colab-superarchives-plugin/blob/master/src/colab_superarchives/views.py#L462

File: src/colab_superarchives/views.py [hunk 2]
```diff
@@ -418,43 +419,86 @@ def get_object(self, *args, **kwargs):
...
+    def filter_lists_by_name(self, user_email, listname):
+        all_lists = mailman.all_lists()
+        lists = []
+        lists_for_address = mailman.mailing_lists(address=user_email,
+                                                  names_only=True)
+        for mlist in all_lists:
+
+            checked = self.check_user_list(mlist, lists_for_address)
+
+            if listname is not None and listname in mlist.get('listname'):
```
Comment (chaws) [inline src/colab_superarchives/views.py:482]: @italopaiva, @Emiliemorais I simplified this if-else to this one here: https://github.com/colab/colab-superarchives-plugin/blob/master/src/colab_superarchives/views.py#L488
hope you guys don't mind : )

Comment (italopaiva) [inline src/colab_superarchives/views.py:482]: Ok! This way is better..

**Comment by chaws** [comment]
Sorry for the delay, this was merged and now it's working fine with pagination. 'add_pagination' was added after filtering lists by listname: https://github.com/colab/colab-superarchives-plugin/blob/master/src/colab_superarchives/views.py#L476

File: src/colab_superarchives/views.py (hunk 1 above)
Comment (italopaiva) [inline src/colab_superarchives/views.py:456]: Well noted, we've missed that..

Comment (chaws) [inline src/colab_superarchives/views.py:456]: no problem...it looks cool now with both the new features

**Pull request merged**